lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -113,6 +115,48 @@ <h3 class="course-title">
                   <a class="enter-course-blocked">${_('View Course')}<span class="sr">&nbsp;${course.display_name_with_default}</span></a>
                 % endif
               % endif
+
+              % if share_settings:
+                <% share_url = urllib.quote_plus(request.build_absolute_uri(course_target)) %>
+                <% share_window_name = 'shareWindow' %>
+                <% share_window_config = 'toolbar=no, location=no, status=no, menubar=no, scrollbars=yes, resizable=yes, width=640, height=480' %>
+                % if share_settings.get('FACEBOOK_SHARING', False):
+                  <% facebook_url = 'https://www.facebook.com/sharer/sharer.php?u=' + share_url %>
+                    <a
+                      data-tooltip="${_('Share on Facebook')}"
+                      class="action action-facebook"
+                      aria-haspopup="true"
+                      aria-expanded="false"
+                      href="${facebook_url}"
+                      target="_blank"
+                      title="${_('Share on Facebook')}"
+                      onclick="window.open('${facebook_url}', '${share_window_name}', '${share_window_config}'); return false;">
+                      <span class="sr">${_('Facebook')}</span>
+                      <i class="fa fa-facebook" aria-hidden="true"></i>
+                    </a>
+                % endif
+                % if share_settings.get('TWITTER_SHARING', False):
+                  <% share_text_default = "I'm learning on {platform_name}:%20".format(platform_name=settings.PLATFORM_NAME) %>
```

Comment:
Actually, I can do the i18n right here in the template -- will fix

On Thu, Apr 2, 2015 at 3:14 PM, Stephen Sanchez <notifications@github.com>
wrote:

> In lms/templates/dashboard/_dashboard_course_listing.html
> <https://github.com/edx/edx-platform/pull/7540#discussion_r27687824>:
>
> > +                  <% facebook_url = 'https://www.facebook.com/sharer/sharer.php?u=' + share_url %>
> > +                    <a
> > +                      data-tooltip="${_('Share on Facebook')}"
> > +                      class="action action-facebook"
> > +                      aria-haspopup="true"
> > +                      aria-expanded="false"
> > +                      href="${facebook_url}"
> > +                      target="_blank"
> > +                      title="${_('Share on Facebook')}"
> > +                      onclick="window.open('${facebook_url}', '${share_window_name}', '${share_window_config}'); return false;">
> > +                      <span class="sr">${_('Facebook')}</span>
> > +                      <i class="fa fa-facebook" aria-hidden="true"></i>
> > +                    </a>
> > +                % endif
> > +                % if share_settings.get('TWITTER_SHARING', False):
> > +                  <% share_text_default = "I'm learning on {platform_name}:%20".format(platform_name=settings.PLATFORM_NAME) %>
>
> I *think* you can change this such that it will get internationalized by
> doing this in settings.py
>
> TWITTER_SHARING_TEXT=_(u"I'm learning on {platform_name}:%20").format(platform_name=settings.PLATFORM_NAME)
>
> You may not want the %20 in there if it gets onto transifex, or add a
> comment, but that'll help reduce code as well as introduce translations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7540/files#r27687824>.
>



-- 
Matthew Drayer
Solutions Architect
edX | MIT
mattdrayer@edx.org